instrumentation/ratpack/ratpack-1.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/ratpack/v1_7/ServerRegistryInstrumentation.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.ratpack.v1_7;
+
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isStatic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import ratpack.handling.HandlerDecorator;
+import ratpack.registry.Registry;
+
+public class ServerRegistryInstrumentation implements TypeInstrumentation {
```

Comment:
This is problematic because ratpack-1.4 has the same instrumentation. Looking at the test output the span is created from the ratpack-1.4 instrumentation and the tests pass when this instrumentation is commented out. Test won't pass when ratpack-1.4 instrumentation is commented out because the span is not created. Usually when there are instrumentations for multiple versions of the same library we make it so that only one of them applies. Here you should decide whether you want to have this instrumentation at all. If you do then we probably need to split the ratpack-1.4 instrumentation into 2 instrumentation modules, one for client and the other for server so we could disable the server one for 1.7+.

Reply:
> Ok, so you're saying here that I can't instrument the same method from 2 different places? one doesn't "wrap" the other?

You can instrument the same method multiple times and one will wrap the other. `InstrumentationModule` implements `Ordered` so you could use https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/805ce0a3cc73cf0d059a870cd0bb88b9edac69cc/instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/instrumentationannotations/AnnotationInstrumentationModule.java#L31-L35 to control the oder of instrumentation (I think instrumentation that is applied first runs later). The nested instrumentation will get suppressed because by default we suppress nested http client spans. This is the same as to how ratpack instrumentation suppresses span from the netty instrumentation. Another alternative that could be considered is using https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/CallDepth.java to skip the code in `injectTracing` for 1.4 if 1.7 is present.

Reply:
Thank you for the learning on this! I've remove this entirely as its duplicative.